pico_w/wifi/access_point/picow_access_point.c

```diff
@@ -10,292 +10,102 @@
-            }
-
-            // Send the body to the client
-            if (con_state->result_len) {
-                err = tcp_write(pcb, con_state->result, con_state->result_len, 0);
-                if (err != ERR_OK) {
-                    DEBUG_printf("failed to write result data %d\n", err);
-                    return tcp_close_client_connection(con_state, pcb, err);
-                }
+            break;
+        }
+        case 1: { // "not_led"
+            if (led_state) {
+                printed = snprintf(pcInsert, iInsertLen, "OFF");
+            } else {
+                printed = snprintf(pcInsert, iInsertLen, "ON");
             }
+            break;
         }
-        tcp_recved(pcb, p->tot_len);
-    }
-    pbuf_free(p);
-    return ERR_OK;
-}
-
-static err_t tcp_server_poll(void *arg, struct tcp_pcb *pcb) {
-    TCP_CONNECT_STATE_T *con_state = (TCP_CONNECT_STATE_T*)arg;
-    DEBUG_printf("tcp_server_poll_fn\n");
-    return tcp_close_client_connection(con_state, pcb, ERR_OK); // Just disconnect clent?
-}
-
-static void tcp_server_err(void *arg, err_t err) {
-    TCP_CONNECT_STATE_T *con_state = (TCP_CONNECT_STATE_T*)arg;
-    if (err != ERR_ABRT) {
-        DEBUG_printf("tcp_client_err_fn %d\n", err);
-        tcp_close_client_connection(con_state, con_state->pcb, err);
-    }
-}
-
-static err_t tcp_server_accept(void *arg, struct tcp_pcb *client_pcb, err_t err) {
-    TCP_SERVER_T *state = (TCP_SERVER_T*)arg;
-    if (err != ERR_OK || client_pcb == NULL) {
-        DEBUG_printf("failure in accept\n");
-        return ERR_VAL;
-    }
-    DEBUG_printf("client connected\n");
-
-    // Create the state for the connection
-    TCP_CONNECT_STATE_T *con_state = calloc(1, sizeof(TCP_CONNECT_STATE_T));
-    if (!con_state) {
-        DEBUG_printf("failed to allocate connect state\n");
-        return ERR_MEM;
-    }
-    con_state->pcb = client_pcb; // for checking
-    con_state->gw = &state->gw;
-
-    // setup connection to client
-    tcp_arg(client_pcb, con_state);
-    tcp_sent(client_pcb, tcp_server_sent);
-    tcp_recv(client_pcb, tcp_server_recv);
-    tcp_poll(client_pcb, tcp_server_poll, POLL_TIME_S * 2);
-    tcp_err(client_pcb, tcp_server_err);
-
-    return ERR_OK;
-}
-
-static bool tcp_server_open(void *arg, const char *ap_name) {
-    TCP_SERVER_T *state = (TCP_SERVER_T*)arg;
-    DEBUG_printf("starting server on port %d\n", TCP_PORT);
-
-    struct tcp_pcb *pcb = tcp_new_ip_type(IPADDR_TYPE_ANY);
-    if (!pcb) {
-        DEBUG_printf("failed to create pcb\n");
-        return false;
-    }
-
-    err_t err = tcp_bind(pcb, IP_ANY_TYPE, TCP_PORT);
-    if (err) {
-        DEBUG_printf("failed to bind to port %d\n",TCP_PORT);
-        return false;
-    }
-
-    state->server_pcb = tcp_listen_with_backlog(pcb, 1);
-    if (!state->server_pcb) {
-        DEBUG_printf("failed to listen\n");
-        if (pcb) {
-            tcp_close(pcb);
+        default: { // unknown tag
+            printed = 0;
```

Comment:
Don't _really_ need this line, given that `printed` is initialised to `0` on line 65 anyway.